api/filter/accesslog.proto

```diff
@@ -2,31 +2,59 @@ syntax = "proto3";
 
 package envoy.api.v2.filter;
 
+import "api/address.proto";
 import "api/base.proto";
 
 import "google/protobuf/duration.proto";
 import "google/protobuf/struct.proto";
 import "google/protobuf/timestamp.proto";
 import "google/protobuf/wrappers.proto";
 
-message AccessLogEntry {
-  // The HTTP request method (RFC 7231/2616)
-  RequestMethod request_method = 1;
+// Envoy access logs describe incoming interaction with Envoy over a fixed
+// period of time, and typically cover a single request/response exchange,
+// (e.g. HTTP), stream (e.g. over HTTP/gRPC), or proxied connection (e.g. TCP).
+// Except where explicitly declared otherwise, all fields describe
+// *downstream* interaction between Envoy and a connected client.
+// Fields describing *upstream* interaction will explicitly include `upstream`
+// in their name.
+message AccessLogCommon {
+  // This field indicates the rate at which this log entry was sampled.
+  // Valid range is (0.0, 1.0].
+  double sample_rate = 1;
+
+  // This field is the IP and port on which the request from the user was
+  // received.
+  SocketAddress destination_host = 2;
+
+  // Properties of the TLS connection, if used
+  TLSProperties tls_properties = 3;
 
   // The time that Envoy started servicing this request
-  google.protobuf.Timestamp start_time = 2;
+  google.protobuf.Timestamp start_time = 4;
 
-  // Incoming protocol variation spoken
-  enum Protocol {
-    PROTOCOL_UNSPECIFIED = 0;
-    HTTP10 = 1;
-    HTTP11 = 2;
-    HTTP2 = 3;
-  }
-  Protocol protocol_variant = 3;
+  // Interval between the first downstream byte received and the last
+  // downstream byte received (i.e. time it takes to receive a request).
+  google.protobuf.Duration time_to_last_byte = 5;
+
+  // Interval between the first downstream byte received and the first upstream
+  // byte received (i.e. time it takes to start receiving a response).
+  google.protobuf.Duration time_to_first_upstream_byte = 6;
+
+  // Interval between the first downstream byte received and the last upstream
+  // byte received (i.e. time it takes to receive a complete response).
+  google.protobuf.Duration time_to_last_upstream_byte = 7;
 
-  // This enum defines the various things that may have occurred while
-  // processing a request.
+  // The (primary) upstream host that handles this exchange.
+  SocketAddress upstream_host = 8;
+
+  // The upstream cluster the upstream_host belongs to.
+  string upstream_cluster = 9;
+
+  // The amount of time it took the upstream server to service the request,
```

Comment:
I just noticed this. Where is this coming from exactly? If this is reported by upstream, how?

Reply:
It's meant to be `X-ENVOY-UPSTREAM-SERVICE-TIME` from the response header, which was part of the default format for text based access logs.

Reply:
I think we can drop this then. This is essentially TTFUB - TTLDB?

Reply:
I'm fine with dropping it. I only included it previously because it was part of the default text access log format.

Reply:
@goaway this still needs to be removed.

Reply:
I thought it was useful in that it allows one to see time spent upstream (as reported by an upstream) relative to that which was spent locally or in transport. It's not quite the same as when the local envoy sends or receives upstream bytes. Do you feel it's too ambiguous? Or just doesn't provide enough additional value?

Reply:
This is not implemented yet. I would log a feature request for this and we can add it later? I would remove from here for now.

Reply:
Alright, sounds good.